### ISSUE TITLE
Project owner/creator enable project monitoring not success

### ISSUE THREAD

**Issue by orangedeng** [rancher/rancher#19381]
With master 4/4

**Steps to reproduce:**
- Create a standard user and set it as cluster member.
- Login to that new user and create a project.
- enable the project monitoring in that project.

**Result:**
- Enabling monitoring success, and you can see project monitoring app in project apps pages. But the app is in install status and does not get deployed.

**Root Cause:**
Logs in Rancher server:
```
[main] 2019/04/04 10:30:07 Starting Tiller v2.10.0-rancher9 (tls=false)
[main] 2019/04/04 10:30:07 GRPC listening on :56577
[main] 2019/04/04 10:30:07 Probes listening on :55311
[main] 2019/04/04 10:30:07 Storage driver is ConfigMap
[main] 2019/04/04 10:30:07 Max history per release is 0
[tiller] 2019/04/04 10:30:07 getting history for release project-monitoring
[storage] 2019/04/04 10:30:07 getting release history for "project-monitoring"
[tiller] 2019/04/04 10:30:07 preparing update for project-monitoring
[storage] 2019/04/04 10:30:07 getting deployed releases from "project-monitoring" history
[storage] 2019/04/04 10:30:08 getting last revision of "project-monitoring"
[storage] 2019/04/04 10:30:08 getting release history for "project-monitoring"
[storage] 2019/04/04 10:30:08 getting release history for "project-monitoring"
[tiller] 2019/04/04 10:30:08 name project-monitoring exists but is not in use, reusing name
[tiller] 2019/04/04 10:30:08 rendering rancher-monitoring chart using values
2019/04/04 10:30:08 [ERROR] ProjectController local/p-wblr4 [project-monitoring-handler] failed with : unable to sync Project p-wblr4(myproject) of Cluster local(local): failed to detect the installation status of monitoring components: [Prometheus StatefulSet isn't deployed, Grafana Deployment isn't deployed]
2019/04/04 10:30:08 info: manifest "rancher-monitoring/templates/servicemonitor.yaml" is empty. Skipping.
2019/04/04 10:30:08 info: manifest "rancher-monitoring/templates/metrics-service.yaml" is empty. Skipping.
2019/04/04 10:30:08 info: manifest "rancher-monitoring/charts/grafana/templates/rbac.yaml" is empty. Skipping.
2019/04/04 10:30:08 info: manifest "rancher-monitoring/charts/grafana/templates/pvc.yaml" is empty. Skipping.
2019/04/04 10:30:08 info: manifest "rancher-monitoring/templates/rbac.yaml" is empty. Skipping.
2019/04/04 10:30:08 info: manifest "rancher-monitoring/templates/deployment.yaml" is empty. Skipping.
[storage] 2019/04/04 10:30:08 updating release "project-monitoring.v1"
[tiller] 2019/04/04 10:30:08 executing 0 pre-delete hooks for project-monitoring
[tiller] 2019/04/04 10:30:08 hooks complete for pre-delete project-monitoring
[kube] 2019/04/04 10:30:08 Starting delete for "expose-grafana-metrics" Service
[kube] 2019/04/04 10:30:08 services "expose-grafana-metrics" not found
[kube] 2019/04/04 10:30:08 Starting delete for "access-grafana" Service
[kube] 2019/04/04 10:30:08 services "access-grafana" not found
[kube] 2019/04/04 10:30:08 Starting delete for "expose-prometheus-metrics" Service
[kube] 2019/04/04 10:30:08 services "expose-prometheus-metrics" not found
[kube] 2019/04/04 10:30:08 Starting delete for "access-prometheus" Service
[kube] 2019/04/04 10:30:08 services "access-prometheus" not found
[kube] 2019/04/04 10:30:08 Starting delete for "grafana-project-monitoring" Deployment
[kube] 2019/04/04 10:30:08 Using reaper for deleting "grafana-project-monitoring"
[kube] 2019/04/04 10:30:08 deployments.extensions "grafana-project-monitoring" not found
[kube] 2019/04/04 10:30:08 Starting delete for "prometheus-project-monitoring" RoleBinding
[kube] 2019/04/04 10:30:08 rolebindings.rbac.authorization.k8s.io "prometheus-project-monitoring" not found
[kube] 2019/04/04 10:30:08 Starting delete for "prometheus-project-monitoring" Role
[kube] 2019/04/04 10:30:08 roles.rbac.authorization.k8s.io "prometheus-project-monitoring" not found
[kube] 2019/04/04 10:30:08 Starting delete for "prometheus-project-monitoring-cattle-prometheus-p-wblr4" ClusterRoleBinding
2019/04/04 10:30:08 uninstall: Failed deletion of "project-monitoring": clusterrolebindings.rbac.authorization.k8s.io "prometheus-project-monitoring-cattle-prometheus-p-wblr4" is forbidden: User "u-58lt9" cannot delete resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope
[kube] 2019/04/04 10:30:08 Starting delete for "p-wblr4-namespaces-readonly-additional-binding" ClusterRoleBinding
2019/04/04 10:30:08 uninstall: Failed deletion of "project-monitoring": clusterrolebindings.rbac.authorization.k8s.io "p-wblr4-namespaces-readonly-additional-binding" is forbidden: User "u-58lt9" cannot delete resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope
[kube] 2019/04/04 10:30:08 Starting delete for "prometheus-project-monitoring-cattle-prometheus-p-wblr4" ClusterRole
2019/04/04 10:30:08 uninstall: Failed deletion of "project-monitoring": clusterroles.rbac.authorization.k8s.io "prometheus-project-monitoring-cattle-prometheus-p-wblr4" is forbidden: User "u-58lt9" cannot delete resource "clusterroles" in API group "rbac.authorization.k8s.io" at the cluster scope
[kube] 2019/04/04 10:30:08 Starting delete for "project-monitoring" ServiceAccount
[kube] 2019/04/04 10:30:08 serviceaccounts "project-monitoring" not found
[kube] 2019/04/04 10:30:08 Starting delete for "grafana-project-monitoring-dashboards" ConfigMap
[kube] 2019/04/04 10:30:09 configmaps "grafana-project-monitoring-dashboards" not found
[kube] 2019/04/04 10:30:09 Starting delete for "grafana-project-monitoring-nginx" ConfigMap
[kube] 2019/04/04 10:30:09 configmaps "grafana-project-monitoring-nginx" not found
[kube] 2019/04/04 10:30:09 Starting delete for "grafana-project-monitoring-provisionings" ConfigMap
[kube] 2019/04/04 10:30:09 configmaps "grafana-project-monitoring-provisionings" not found
[kube] 2019/04/04 10:30:09 Starting delete for "prometheus-project-monitoring-nginx" ConfigMap
[kube] 2019/04/04 10:30:09 configmaps "prometheus-project-monitoring-nginx" not found
[kube] 2019/04/04 10:30:09 Starting delete for "prometheus-project-monitoring-additional-scrape-configs" Secret
[kube] 2019/04/04 10:30:09 secrets "prometheus-project-monitoring-additional-scrape-configs" not found
[kube] 2019/04/04 10:30:09 Starting delete for "prometheus-project-monitoring-additional-alertmanager-configs" Secret
[kube] 2019/04/04 10:30:09 secrets "prometheus-project-monitoring-additional-alertmanager-configs" not found
[kube] 2019/04/04 10:30:09 Starting delete for "project-monitoring" Prometheus
2019/04/04 10:30:09 uninstall: Failed deletion of "project-monitoring": prometheuses.monitoring.coreos.com "project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "prometheuses" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"
[kube] 2019/04/04 10:30:09 Starting delete for "grafana-project-monitoring" ServiceMonitor
2019/04/04 10:30:09 uninstall: Failed deletion of "project-monitoring": servicemonitors.monitoring.coreos.com "grafana-project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"
[kube] 2019/04/04 10:30:09 Starting delete for "prometheus-project-monitoring" ServiceMonitor
2019/04/04 10:30:09 uninstall: Failed deletion of "project-monitoring": servicemonitors.monitoring.coreos.com "prometheus-project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"
[storage] 2019/04/04 10:30:09 updating release "project-monitoring.v1"
[tiller] 2019/04/04 10:30:09 error: clusterrolebindings.rbac.authorization.k8s.io "prometheus-project-monitoring-cattle-prometheus-p-wblr4" is forbidden: User "u-58lt9" cannot delete resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope
[tiller] 2019/04/04 10:30:09 error: clusterrolebindings.rbac.authorization.k8s.io "p-wblr4-namespaces-readonly-additional-binding" is forbidden: User "u-58lt9" cannot delete resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope
[tiller] 2019/04/04 10:30:09 error: clusterroles.rbac.authorization.k8s.io "prometheus-project-monitoring-cattle-prometheus-p-wblr4" is forbidden: User "u-58lt9" cannot delete resource "clusterroles" in API group "rbac.authorization.k8s.io" at the cluster scope
[tiller] 2019/04/04 10:30:09 error: prometheuses.monitoring.coreos.com "project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "prometheuses" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"
[tiller] 2019/04/04 10:30:09 error: servicemonitors.monitoring.coreos.com "grafana-project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"
[tiller] 2019/04/04 10:30:09 error: servicemonitors.monitoring.coreos.com "prometheus-project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"
2019/04/04 10:30:09 [ERROR] AppController p-wblr4/project-monitoring [helm-controller] failed with : failed to install app project-monitoring. Error: UPGRADE FAILED: Upgrade --force successfully deleted the previous release, but encountered 6 error(s) and cannot continue: clusterrolebindings.rbac.authorization.k8s.io "prometheus-project-monitoring-cattle-prometheus-p-wblr4" is forbidden: User "u-58lt9" cannot delete resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope; clusterrolebindings.rbac.authorization.k8s.io "p-wblr4-namespaces-readonly-additional-binding" is forbidden: User "u-58lt9" cannot delete resource "clusterrolebindings" in API group "rbac.authorization.k8s.io" at the cluster scope; clusterroles.rbac.authorization.k8s.io "prometheus-project-monitoring-cattle-prometheus-p-wblr4" is forbidden: User "u-58lt9" cannot delete resource "clusterroles" in API group "rbac.authorization.k8s.io" at the cluster scope; prometheuses.monitoring.coreos.com "project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "prometheuses" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"; servicemonitors.monitoring.coreos.com "grafana-project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"; servicemonitors.monitoring.coreos.com "prometheus-project-monitoring" is forbidden: User "u-58lt9" cannot delete resource "servicemonitors" in API group "monitoring.coreos.com" in the namespace "cattle-prometheus-p-wblr4"
```

And the log means the project creator/owner don't have access to the `monitoring.coreos.com` group resources and `rbac.authorization.k8s.io` group resources(`clusterrole` and `clusterrolebinding`).

**Comment by jiaqiluo** [comment]
the bug is reproduced in Rancher v2.2.1

**Comment by loganhz** [comment]
Waiting for https://github.com/rancher/rancher/pull/19766 https://github.com/rancher/types/pull/801 https://github.com/rancher/system-charts/pull/37 merged and https://github.com/rancher/rancher/pull/19766 https://github.com/rancher/types/pull/801 backport to 2.2. branch.

**Comment by jiaqiluo** [comment]
The validation fails because Rancher cannot find `system-library-rancher-monitoring-0.0.3` in the `
system-library` catalog

Error message 
````
2019/05/07 18:15:02 [ERROR] AppController p-s8sws/cluster-monitoring [helm-controller] failed with :
catalogtemplateversions.management.cattle.io "system-library-rancher-monitoring-0.0.3" not found
```
@loganhz

**Comment by jiaqiluo** [comment]
The working around here is to set the branch to `dev` for the `system-library`

--------

The bug fix is validated on Rancher: master e1e6fc93

Step:
- as the admin, provision an EC2 cluster 
- as the admin, create a standard user and set it as the cluster member
- login as the new user and create a project.
- enable the project monitoring in that project.

Result:
- the project monitoring is up 
- no errors showing in the log
